pkg/controllers/vdb/podfacts.go

```diff
@@ -1204,3 +1204,43 @@ func (p *PodFacts) FindUnsandboxedSubclustersStillInSandboxStatus(scSbInVdbStatu
 	}
 	return sbScMap
 }
+
+// FindSecondarySubclustersWithDifferentImage will scan the secondary subclusters in main cluster
+// and return the secondary subclusters that have different vertica image than main cluster
+func (p *PodFacts) FindSecondarySubclustersWithDifferentImage() []string {
```

Comment:
Can this be a general purpose function? A function that gets a an image an return a list of secondary subclusters not running that image. And the caller can pass in vdb.spec.Image.

Reply:
I think the pod facts have the most accurate image value. It's possible vdb.spec.Image is different than the one in a primary subcluster when the main cluster is under upgrade.

Reply:
I see. Then that also means we should not be using vdb.spec.image to set the image in the statefulset's spec like it is done in that reconciler.